src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectCameraInput.cpp

```diff
@@ -0,0 +1,401 @@
+            // the body index processing thread has fallen behind.
+            OutputDebugString(L"Warning: frame buffer is completely full, and we can't begin writing to the next frame");
+            continue;
+        }
+
+        k4a_capture_t capture = nullptr;
+
+        switch (k4a_device_get_capture(_k4aDevice, &capture, K4A_WAIT_INFINITE))
+        {
+        case K4A_WAIT_RESULT_TIMEOUT:
+            OutputDebugString(L"Error: Timed out waiting for AzureKinect capture");
+            return;
+        case K4A_WAIT_RESULT_FAILED:
+            OutputDebugString(L"Error: Failed to capture from AzureKinect");
+            return;
+        }
+
+        auto colorImage = k4a_capture_get_color_image(capture);
+        if (colorImage != nullptr)
+        {
+            _colorImageStride = k4a_image_get_stride_bytes(colorImage);
+            _cameraFrames[frameIndex]->StageImage(AzureKinectImageType::Color, colorImage);
+            UpdateArUcoMarkers(colorImage);
+
+            if (_captureDepth)
+            {
+                auto depthImage = k4a_capture_get_depth_image(capture);
+                if (depthImage != nullptr)
+                {
+                    k4a_transformation_depth_image_to_color_camera(_transformation, depthImage, _transformedDepthImage);
+                    _cameraFrames[frameIndex]->StageImage(AzureKinectImageType::Depth, _transformedDepthImage);
+
+#if defined(INCLUDE_AZUREKINECT_BODYTRACKING)
+                    if (_captureBodyMask)
+                    {
+                        // Move to the state of writing the body mask before enqueuing the capture.
+                        // This will put the frame in a state where it's waiting for the body mask
+                        // before entering the Staged state.
+                        _cameraFrames[frameIndex]->BeginWritingBodyMask();
+                        k4a_wait_result_t queue_capture_result = k4abt_tracker_enqueue_capture(_k4abtTracker, capture, K4A_WAIT_INFINITE);
+
+                        if (queue_capture_result == K4A_WAIT_RESULT_FAILED)
+                        {
+                            printf("Error: Adding capture to tracker process queue failed!\n");
+                            _cameraFrames[frameIndex]->EndWritingBodyMask();
+                        }
+                    }
+#endif
+
+                    k4a_image_release(depthImage);
+                }
+            }
+            k4a_image_release(colorImage);
+        }
+        k4a_capture_release(capture);
+
+        // End the writing state. If the state machine already transitioned
+        // to WritingBodyMask, this will have no effect. If the state machine
+        // is still in the WritingColorAndDepth state, it will move to the 
+        // Staged state to mark the frame as ready for reading.
+        _cameraFrames[frameIndex]->EndWritingColorAndDepth();
+        _currentFrameIndex++;
```

Comment:
do we want any sort of wait here or do we just want to keep looping with no pause?